### PR TITLE
Add widget for notofications

### LINKED ISSUE #1774: EXTENSIBILITY: Introduce notification Widget
**Description**

Introduce a notification Widget that can be added to the create/edit form.

**Acceptance Criteria**

- [ ] We offer the possibility to choose any warning/info level
- [ ] The text is translatable
- [ ] can be toggle on or of using jsonata via visibility

**Attachments**

image

### PR DESCRIPTION
**Pull request by mrCherry97**
**Description**

Changes proposed in this pull request:

- add `Alert` widget for details and list,
- add `Alert` to gateways,
- allow JSONata for these widgets,
- fix visibility for widgets without `path` and `var`,
- add info about this to docs.

**Related issue(s)**
Closes #1774

### CODE REVIEW

**Comment by mrCherry97** [comment]
TODO:
- [ ] jsonata,
- [ ] render widget without `path` and `var`,
- [ ] one by one aren't displayed

File: examples/resources/istio/gateways.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 kind: ConfigMap
 apiVersion: v1
 metadata:
-  name: gateways
+  name: cherry-gateways
```
Comment (qbalukom) [inline examples/resources/istio/gateways.yaml:4]: this probably shouldn't be `cherry-gateways`?

File: examples/resources/istio/gateways.yaml [hunk 2]
```diff
@@ -208,9 +215,9 @@ data:
         "group": "networking.istio.io",
         "version": "v1beta1"
       },
-      "urlPath": "ext-gateways",
-      "category": "Istio",
-      "name": "Ext Gateways",
+      "urlPath": "cherry-gateways",
```
Comment (qbalukom) [inline examples/resources/istio/gateways.yaml:218]: same here?

**Review by qbalukom — CHANGES_REQUESTED** [review]

File: docs/extensibility/display-section.md [hunk 3]
```diff
@@ -396,6 +397,32 @@ Text widgets render values as a simple text. This is the default behavior for al
 
 Block widgets are more complex layouts and you must use them only in the details body.
 
+### Alert
+
+Alert widgets display values using predefined types.
+
+#### Widget-specific parameters
+
+- **source** - information what has to be displayed.
```
Comment (qbalukom) [inline docs/extensibility/display-section.md:406]: do we need this mention? `source` is the default for all components...
just an idea. maybe it makes sense to leave it...

File: docs/extensibility/display-section.md [hunk 4]
```diff
@@ -396,6 +397,32 @@ Text widgets render values as a simple text. This is the default behavior for al
...
+- **disableMargin** - an optional boolean which disables the margin outside the alert body.
+- **type** - type of alert:
+  - **information** - Defalut
+  - **warning**
+  - **error**
+  - **success**
```
Comment (qbalukom) [inline docs/extensibility/display-section.md:412]: i'm not sure if listing all the fields as separate entries is necessary. Maybe just do something like this?
```suggestion
- **type** - type of alert. One of **information**, **warning**, **error** or **success**. By default it's set to **information**.
```

File: docs/extensibility/form-section.md [hunk 5]
```diff
@@ -103,6 +104,35 @@ In the example, the visibility for item price and color are analogous - the form
...
 
+### Alert
+
+Alert widgets display information for user using predefined types.
+
+#### Widget-specific parameters
+
+- **alert** - information what has to be displayed.
+- **disableMargin** - an optional boolean which disables the margin outside the alert body.
+- **type** - type of alert:
```
Comment (qbalukom) [inline docs/extensibility/form-section.md:115]: same as details

**Review by qbalukom — CHANGES_REQUESTED** [review]

File: docs/extensibility/display-section.md (hunk 3 above)
Comment (grego952) [inline docs/extensibility/display-section.md:406]:
```suggestion
- **source** - information that has to be displayed.
```
What type of information? Can you provide some more details/examples?

Comment (grego952) [reply docs/extensibility/display-section.md:406]: Is this **source** the same as mentioned in the [Display section](https://github.com/kyma-project/busola/blob/main/docs/extensibility/display-section.md#available-list-section-parameters)? If yes, then please use the same definition.

File: docs/extensibility/display-section.md [hunk 6]
```diff
@@ -396,6 +397,32 @@ Text widgets render values as a simple text. This is the default behavior for al
...
 
+### Alert
+
+Alert widgets display values using predefined types.
+
+#### Widget-specific parameters
+
+- **source** - information what has to be displayed.
+- **disableMargin** - an optional boolean which disables the margin outside the alert body.
+- **type** - type of alert:
```
Comment (grego952) [inline docs/extensibility/display-section.md:408]:
```suggestion
- **type** - specifies one of the alert types: **information**, **warning**, **error**, or **success**. By default, it's set to **information**.
```
As per qbalukom suggestion.

File: docs/extensibility/display-section.md [hunk 7]
```diff
@@ -396,6 +397,32 @@ Text widgets render values as a simple text. This is the default behavior for al
...
+  - **information** - Defalut
+  - **warning**
+  - **error**
+  - **success**
+
+#### Example
+
+```yaml
+- source: "'I am some warning for user'"
```
Comment (grego952) [inline docs/extensibility/display-section.md:417]:
```suggestion
- source: "'I am some warning for a user'"
```

File: docs/extensibility/form-section.md [hunk 8]
```diff
@@ -103,6 +104,35 @@ In the example, the visibility for item price and color are analogous - the form
 
 Form widgets are used in the resource forms.
 
+### Alert
+
+Alert widgets display information for user using predefined types.
```
Comment (grego952) [inline docs/extensibility/form-section.md:109]:
```suggestion
Alert widgets display values using predefined types.
```
Can we use the same definition as for the display section? Just to stay consistent.

File: docs/extensibility/form-section.md [hunk 9]
```diff
@@ -103,6 +104,35 @@ In the example, the visibility for item price and color are analogous - the form
...
+
+#### Widget-specific parameters
+
+- **alert** - information what has to be displayed.
```
Comment (grego952) [inline docs/extensibility/form-section.md:113]: See my previous comment about **alert**.

File: docs/extensibility/display-section.md [hunk 10]
```diff
@@ -394,6 +395,27 @@ Text widgets render values as a simple text. This is the default behavior for al
...
 Block widgets are more complex layouts and you must use them only in the details body.
 
+### Alert
+
+Alert widgets display values using predefined types.
+
+#### Widget-specific parameters
+
+- **disableMargin** - an optional boolean which disables the margin outside the alert body.
+- **type** - type of alert. One of **information**, **warning**, **error** or **success**. By default it's set to **information**.
```
Comment (grego952) [inline docs/extensibility/display-section.md:405]:
```suggestion
- **type** - specifies one of the alert types: **information**, **warning**, **error**, or **success**. By default, it's set to **information**.
```

File: docs/extensibility/form-section.md [hunk 11]
```diff
@@ -103,6 +104,31 @@ In the example, the visibility for item price and color are analogous - the form
...
 
+### Alert
+
+Alert widgets display values using predefined types.
+
+#### Widget-specific parameters
+
+- **source** - information that has to be displayed.
+- **disableMargin** - an optional boolean which disables the margin outside the alert body.
+- **type** - type of alert. One of **information**, **warning**, **error** or **success**. By default it's set to **information**.
```
Comment (grego952) [inline docs/extensibility/form-section.md:115]:
```suggestion
- **type** - specifies one of the alert types: **information**, **warning**, **error**, or **success**. By default, it's set to **information**.

File: docs/extensibility/display-section.md [hunk 12]
```diff
@@ -394,6 +395,27 @@ Text widgets render values as a simple text. This is the default behavior for al
...
 Block widgets are more complex layouts and you must use them only in the details body.
 
+### Alert
+
+Alert widgets display values using predefined types.
+
+#### Widget-specific parameters
+
+- **disableMargin** - an optional boolean which disables the margin outside the alert body.
+- **severity** - specifies one of the alert severity: **information**, **warning**, **error**, or **success**. By default, it's set to **information**.
```
Comment (grego952) [inline docs/extensibility/display-section.md:405]:
```suggestion
- **severity** - specifies one of the alert severities: **information**, **warning**, **error**, or **success**. By default, it's set to **information**.
```

File: docs/extensibility/form-section.md [hunk 13]
```diff
@@ -103,6 +104,31 @@ In the example, the visibility for item price and color are analogous - the form
 
 Form widgets are used in the resource forms.
 
+### Alert
+
+Alert widgets display values using predefined types.
+
+#### Widget-specific parameters
+
+- **alert** - information that has to be displayed.
```
Comment (grego952) [inline docs/extensibility/form-section.md:113]:
```suggestion
- **alert** - information that you want to display.
```

File: docs/extensibility/form-section.md [hunk 14]
```diff
@@ -103,6 +104,31 @@ In the example, the visibility for item price and color are analogous - the form
...
+- **disableMargin** - an optional boolean which disables the margin outside the alert body.
+- **severity** - specifies one of the alert severity: **information**, **warning**, **error**, or **success**. By default, it's set to **information**.
```
Comment (grego952) [inline docs/extensibility/form-section.md:115]:
```suggestion
- **severity** - specifies one of the alert severities: **information**, **warning**, **error**, or **success**. By default, it's set to **information**.
```

**Review by grego952 — APPROVED** [review]

**Review by qbalukom — APPROVED** [review]

**Pull request merged**